src/System.Management.Automation/engine/COM/ComUtil.cs

```
@@ -355,4 +356,74 @@ internal static ComMethodInformation[] GetMethodInformationArray(COM.ITypeInfo t
+        {
+            _element[0] = null;
+            int result = _enumVariant.Next(1, _element, IntPtr.Zero);
+            return result == 0;
+        }
+
+        public void Reset()
+        {
+            _element[0] = null;
+            _enumVariant.Reset();
+        }
+
+        internal static ComEnumerator Create(object comObject)
+        {
+            if (!comObject.GetType().IsCOMObject) { return null; }
+
+            // The passed-in COM object could already be a IEnumVARIANT interface.
+            // e.g. user call '_NewEnum()' on a COM collection interface.
+            var enumVariant = comObject as COM.IEnumVARIANT;
+            if (enumVariant == null)
+            {
+                // The passed-in COM object could be a collection.
+                var enumerable = comObject as IEnumerable;
+                var target = comObject as IDispatch;
+                if (enumerable != null && target != null)
+                {
+                    try
+                    {
+                        var comTypeInfo = ComTypeInfo.GetDispatchTypeInfo(comObject);
+                        if (comTypeInfo != null && comTypeInfo.NewEnumInvokeKind.HasValue)
+                        {
+                            // The COM object is a collection and also a IDispatch interface, so we try to get a
+                            // IEnumVARIANT interface out of it by invoking its '_NewEnum (DispId: -4)' function.
+                            var result = ComInvoker.Invoke(target, ComTypeInfo.DISPID_NEWENUM,
+                                                           args: Utils.EmptyArray<object>(), byRef: null,
+                                                           invokeKind: comTypeInfo.NewEnumInvokeKind.Value);
+                            enumVariant = result as COM.IEnumVARIANT;   
+                        }
+                    }
+                    catch (Exception)
+                    {
+                        /* Catch all exception. */
```

Obvious comment and typo (exception -> exception**s**). Catch all exceptions - why? Please enhance the comment or remove.